src/Disciplina/DB/DSL/Interpret.hs

```diff
@@ -0,0 +1,99 @@
+module Disciplina.DB.DSL.Interpret
+              ( runQuery
+              , RunQuery (..)
+              )
+              where
+
+import Universum
+import Disciplina.DB.DSL.Types (QueryTx(..), QueryTxs(..), WHERE (..)
+                               ,TxIdEq (..), TxGrade(..), QueryObj(..)
+                               ,TxsFilterExpr(..), ObjHashEq(..))
+import Disciplina.Educator.Txs (PrivateTxId(..), PrivateTx(..))
+import Disciplina.Crypto (Hash)
+import qualified Disciplina.Core as Core (Grade(..), SubjectId(..))
+
+class (Monad m) => MonadSearchTxObj m where
+  runTxQuery :: QueryTx -> m (Maybe PrivateTx)
+  runTxsQuery :: QueryTxs -> m [PrivateTx]
+  runObjQuery :: QueryObj -> m (Maybe Int)
+
+instance MonadSearchTxObj IO where
+  runTxQuery = runIOTxQuery
+  runTxsQuery = runIOTxsQuery
+  runObjQuery = runIOObjQuery
+
+-- | Query type determine query return type
+class RunQuery a b | a -> b where
```

Comment:
Thinking about it, wouldn't it be better to have `class  (Monad m) => RunQuery m a b | a -> b` and skip `MonadSearchTxObj` type class?

Reply:
No, I don't think so. This way you'll have to declare three separate `RunQuery` instances for each separate monad, instead of three `RunQuery` instances in total.